runtime/src/payout.rs

```diff
@@ -0,0 +1,208 @@
+use super::*;
+pub use sp_runtime::{Percent, Perquintill};
+
+fn abs(v: f64) -> f64 {
+    if v > 0f64 {
+        v
+    } else {
+        -v
+    }
+}
+
+fn exp(p: f64) -> f64 {
+    let mut res: f64 = 0f64;
+    // Sums: 1 + x + x^2/2! + x^3/3! + x^4/4! + x^5/5! +
```

Comment:
`p` is always lower than 1. For the negative values we can just set a cut off when the current staking is less `1/21` of the target one. In this case the _correction_ of the target inflation is since `C*0.000000002` percent => we can fall to use the base inflation.